### PR TITLE
Adjust delta lake product tests to collect statistics on write changes

### PR DESCRIPTION
**Pull request by pajaks**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
/test-with-secrets sha=2209765a1275c491f31208d1f2cb8fe89adc43ea

**Comment by ebyhr** [comment]
/test-with-secrets sha=798a89ee9deb4c2ca30cf7344a6fcdcab96edc15

**Comment by pajaks** [comment]
CI hit: https://github.com/trinodb/trino/issues/13199

**Review by homar — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCheckpointsCompatibility.java [hunk 1]
```diff
@@ -498,21 +498,22 @@ private void testWriteStatsAsStructDisabled(Consumer<String> sqlExecutor, String
...
     public void testDatabricksWriteStatsAsJsonEnabled(String type, String inputValue, Double nullsFraction, Object statsValue)
     {
         String tableName = "test_dl_checkpoints_write_stats_as_json_enabled_databricks_" + randomNameSuffix();
-        testWriteStatsAsJsonEnabled(sql -> onDelta().executeQuery(sql), tableName, "default." + tableName, type, inputValue, nullsFraction, statsValue);
+        testWriteStatsAsJsonEnabled(sql -> onDelta().executeQuery(sql), tableName, "default." + tableName, type, inputValue, null, null, nullsFraction, statsValue);
     }
 
-    private void testWriteStatsAsJsonEnabled(Consumer<String> sqlExecutor, String tableName, String qualifiedTableName, String type, String inputValue, Double nullsFraction, Object statsValue)
+    private void testWriteStatsAsJsonEnabled(Consumer<String> sqlExecutor, String tableName, String qualifiedTableName, String type, String inputValue,
+            Double dataSize, Double distinctValues, Double nullsFraction, Object statsValue)
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCheckpointsCompatibility.java:516]: Please don't wrap arguments on the way. The convention is a single line or wrapping per argument.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**